src/routes/report.tsx

```diff
@@ -11,11 +20,30 @@ export interface ReportRouteProps {
   env?: string;
 }
 
-function ReportRoute(props: ReportRouteProps): VNode {
+function ReportRoute({
+  path,
+  start: startParam,
+  finish: finishParam,
+  ial: ialParam,
+  agency,
+  env: envParam,
+}: ReportRouteProps): VNode {
+  const endOfPreviousWeek = utcWeek.floor(new Date());
+  const startOfPreviousWeek = utcWeek.floor(new Date(endOfPreviousWeek.valueOf() - 1));
+
+  const start = (startParam && yearMonthDayParse(startParam)) || startOfPreviousWeek;
+  const finish = (finishParam && yearMonthDayParse(finishParam)) || endOfPreviousWeek;
+  const ial = (parseInt(ialParam || "", 10) || DEFAULT_IAL) as 1 | 2;
+  const env = envParam || DEFAULT_ENV;
+  const [allAgencies, setAllAgencies] = useState([] as string[]);
+
   return (
-    <ReportFilterControls {...props}>
+    <ReportFilterControlsContext.Provider
+      value={{ path, start, finish, ial, agency, env, setAllAgencies }}
+    >
+      <ReportFilterControls agencies={allAgencies} />
```

Comment:
The handling of "agencies" feels a bit fragmented, where we store the source of truth in the route, pass the state updater to the context provider, and the current value to the filter controls.

I'd wonder if the context should be handling all of this, and making both `setAllAgencies` and `agencies` available through the context.

e.g.

```tsx
function ReportFilterControlsContextProvider({
  start,
  finish,
  ial,
  setAllAgencies,
  env,
  children,
}: ReportFilterControlsContextProviderProps) {
  const start = (startParam ? yearMonthDayParse(startParam) : null) || startOfPreviousWeek;
  const finish = (finishParam ? yearMonthDayParse(finishParam) : null) || endOfPreviousWeek;
  const ial = (parseInt(ialParam || "", 10) || DEFAULT_IAL) as 1 | 2;
  const env = envParam || DEFAULT_ENV;
  const [allAgencies, setAllAgencies] = useState([] as string[]);

  return (
    <ReportFilterControlsContext.Provider
      value={{ start, finish, ial, env, allAgencies, setAllAgencies }}
    >
      {children}
    </ReportFilterControlsContext.Provider>
  );
}
```

(This might also help for what you mentioned at https://github.com/18F/identity-reporting/pull/21#discussion_r709490268 )

Optionally, we could go even further to have a separate context just for managing "agencies", since it feels a bit like it's not explicitly tied to filtering.

e.g.

```tsx
<AgenciesContextProvider>
  <ReportFilterControlsContextProvider start={start} finish={finish} ial={ial} agency={agency} env={env}>
    <ReportFilterControls />
    <DailyAuthsReport />
  </ReportFilterControlsContextProvider>
</AgenciesContextProvider>
```

Reply:
I like the idea of the separate agencies context, will give that a shot

Reply:
added AgenciesContext in c69df8e919276d801e7eed83259041aa261420aa